### DISCUSSION TITLE
Fanpico High Power Variant

### DISCUSSION THREAD

**Discussion by siodor** [tjko/fanpico#95] [Ideas]
Hello @tjko 

**Feature Request:**
Make a high power version of Fanpico for fans such as Delta Electronics PFR1212DHE-SP00, 120x120x38 PWM fan with 12VDC @ 44.4W with max 7400 RPM.

**Explanation**
I'm building a custom sound dampened 19" rack which also has to withstand some harsh environmental conditions. For this reason I'm planning to add air handling unit (AHU) pocket filters (F9 @ EN 779, ePM1 90% @ ISO 16890) which normally run with large AHU fans and due to the high grade of filtering add quite some resistance to the air flow. To keep things compact and with parts I can source myself I'm planning to add proper industrial 120mm fans.

**Comment by tjko** [comment]
A board variant that supports high power (current) fans, certainly is doable.  But question is how much demand there would be for one? Since it likely would add significant cost (in low quantities) to produce PCB with thicker copper (and additional layers, etc).

I wonder if better approach would be to use separate "power board" for powering fans that require high current. Board could have some commonly used (high) power input connectors and then connectors for fans and also connectors to connect to fanpico (fan ports). This way fanpico board would only control/monitor the fans and not provide power to fans.  And board might be useful in other applications as well, if want to replace existing fan with much higher power one...


Another alternative would be to build "splitter" cable adapters, that have one female fan connector (for the high power fan) and two fan connectors, one to connect to a power supply and one to connect to FanPico fan port. Something like this:


<img width="566" alt="Screenshot 2024-07-28 at 11 36 50" src="https://github.com/user-attachments/assets/78e53e56-c403-4926-9bb1-cca50937a31a">

**Comment by tjko** [comment]
Here is quick draft of a "power board" for FanPico. This should allow delivering about 120W of power to 1-4 fans, one or more boards could be used as needed.

Board could be powered either using a PC "Molex" power connector or directly wiring to a terminal block.

Board could be populated either with a 5x20mm fuse holder or 2920 SMD poly fuse. There is also optional footprints for addin power indicator led and auxiliary output terminal block...

![fanpico-powerboard](https://github.com/user-attachments/assets/83f8eb8b-e3c4-44f3-b178-073d394e0243)

This should also allow running fans that require different voltage than 12V...  (5-24V range should be "safe", but if going above 30V then there is risk damaging tachometer input on FanPico if fan has internal pull-up on tachometer pin).

**Reply by siodor** [reply]
Hello @tjko 

This looks great!

Might I please suggest to add series resistor and a Zener diode to the TACHO signal going to FanPico to limit the voltage returning from higher voltage fans?

Basically what's described here: https://electronics.stackexchange.com/questions/81429/why-do-we-connect-a-resistor-before-a-zener-diode

Vin is the TACHO signal coming from the fan and Vout is the voltage limited TACHO signal going to FanPico.

**Reply by tjko** [reply]
@siodor,  FanPico already has "blocking" diodes on Tachometer pins, since some fans (or fan manufacturers) have pull up to Vcc inside the fan, while Intel spec calls for fan to have open collector output (and motherboard to have the pull-up). So RP2040 only sees 3.3V tachometer signal...

I think current diodes (BAT54C) have 30V breakdown voltage, so it should be safe to use up to 24V fans.

I guess could add additional protection diodes on this board if there is need for those?   Are higher voltage (48V, etc) DC fans common that can be still controlled with 5V PWM signal?
